### ISSUE TITLE
push zero length vector

### ISSUE THREAD

**Issue by daroczig** [wlandau/txtq#15]
It seems that calling `clean()` on a queue can cause problems with indexing:

```r
library(txtq)
t <- tempfile()
q <- txtq(t)

for (i in 1:3) q$push(title=i, message=i)
for (i in 1:3) {
    cat(q$pop(1)$title, sep = '\n')
}
#> 1
#> 2
#> 3

q$clean()

for (i in 1:3) q$push(title=i, message=i)
for (i in 1:3) {
    cat(q$pop(1)$title, sep = '\n')
}
#> 1
#> 1
#> 2
```

**Comment by daroczig** [comment]
IMO this is due to `push`-ing vectors with the length of zero:

```r
library(txtq)
t <- tempfile()
q <- txtq(t)
q$push(title = character(0), message = character(0))

for (i in 1:3) q$push(title=i, message=i)
for (i in 1:3) {
    cat(q$pop(1)$title, sep = '\n')
}
#> 1
#> 1
#> 2
```

**Comment by daroczig** [comment]
And the above fitered down to 

```r
> length(out)
[1] 0

> paste(out, collapse = "\n")
[1] ""
```

in `txtq_push`, so increasing by zero but adding an empty string to the DB

**Comment by wlandau** [comment]
Fixed via #16.
